streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1248,15 +1248,16 @@ protected StreamTask createStreamTask(final TaskId id, final Collection<TopicPar
     }
 
     private Producer<byte[], byte[]> createProducer(final TaskId id) {
-        final Map<String, Object> producerConfigs = config.getProducerConfigs(threadClientId);
```

Comment:
If metric names changes of course, as we get a producer metric per task instead of per thread. However, as the threadID has some generated UUID in it anyway, any automatic tool receiving the metric must be able to handle that case anyway as metric name change for rolling bounce even for non-EOS case. So it might not be too bad.